src/ripple/core/JobCoro.ipp

```diff
@@ -41,21 +42,45 @@ JobCoro::JobCoro (detail::JobCoro_create_t, JobQueue& jq, JobType type,
 
 inline
 void
-JobCoro::yield () const
+JobCoro::yield() const
 {
     (*yield_)();
 }
 
 inline
 void
-JobCoro::post ()
+JobCoro::post()
 {
+    {
+        std::lock_guard<std::mutex> lk(mutex_run_);
+        running_ = true;
+    }
+
     // sp keeps 'this' alive
     jq_.addJob(type_, name_,
         [this, sp = shared_from_this()](Job&)
         {
-            std::lock_guard<std::mutex> lock (mutex_);
+            auto saved = detail::getLocalValues().release();
+            detail::getLocalValues().reset(&lvs_);
+            std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
Apparently thread locals break my brain. Got it. So the `mutex_` protects `coro_` only? If that's the case, how about a block with just `lock` and the `coro_` call in scope?